hub/api/tensor.py

```diff
@@ -1,35 +1,68 @@
 from hub.util.slice import merge_slices
 import numpy as np
 from typing import Union
+from hub.core.storage.provider import StorageProvider
+from hub.core.storage.memory import MemoryProvider
+from hub.core.storage.local import LocalProvider
+from hub.core.chunk_engine.read import read_array, read_tensor_meta
+from hub.core.chunk_engine.write import write_array, write_tensor_meta
 
 
 class Tensor:
-    def __init__(self, uuid: str, tensor_slice: slice = slice(None)):
+    def __init__(
+        self,
+        key: str,
+        provider: StorageProvider,
+        tensor_slice: slice = slice(None),
+    ):
         """Initialize a new tensor.
 
         Note:
             This operation does not create a new tensor in the backend,
```

Comment:
@imshashank made a point that we should not call the storage a "backend". should say "in the storage provider"

Reply:
Yep.